### PR TITLE
Preference to Follow HTTP Recirects

### PR DESCRIPTION
**Pull request by grahamgilbert**
This adds a preference to allow Munki to follow HTTP redirects (``FollowHTTPRedirects``). 

The possible values are:

* ``none`` - This is the default and is the same as the present behaviour. No redirects are followed.
* ``https`` - Only redirects to URLs using HTTPS are followed. Redirects to HTTP URLs are not followed.
* ``all`` - Redirects to both HTTP and HTTPS URLs are followed.

### CODE REVIEW

File: code/client/munkilib/munkicommon.py [hunk 1]
```diff
@@ -1212,7 +1212,8 @@ def pref(pref_name):
         'SuppressUserNotification': False,
         'SuppressAutoInstall': False,
         'SuppressStopButtonOnInstall': False,
-        'PackageVerificationMode': 'hash'
+        'PackageVerificationMode': 'hash',
+        'FollowTTPRedirects': 'none',
```
Comment (keeleysam) [inline code/client/munkilib/munkicommon.py]: Missed an H here, should be FollowHTTPRedirects

Comment (grahamgilbert) [inline code/client/munkilib/munkicommon.py]: Well that's embarrassing. I must have sausage fingered it just before committing.

**Comment by grahamgilbert** [comment]
Is there anything else I can do to help with getting this merged in?

**Comment by gregneagle** [comment]
I'll need you to write some documentation and post an announcement to munki-dev and munki-discuss explaining how to use the functionality, what are the expected behaviors, and how to test and provide feedback.

**Comment by gregneagle** [comment]
This probably should be part of a Munki 2.4 release.

**Comment by grahamgilbert** [comment]
No problem, I'll do that tomorrow.

**Pull request merged**